modules/internal/src/main/scala/higherkindness/mu/rpc/internal/serviceImpl.scala

```diff
@@ -338,47 +386,229 @@ object serviceImpl {
+          case _           => TermName("POST")
+        }
+
+        val executionClient: Tree = response match {
+          case MonixObservableTpe(_, _) =>
+            q"Observable.fromReactivePublisher(client.stream(request).flatMap(_.asStream[${response.safeInner}]).toUnicastPublisher)"
+          case Fs2StreamTpe(_, _) =>
+            q"client.stream(request).flatMap(_.asStream[${response.safeInner}])"
+          case _ =>
+            q"client.expectOr[${response.safeInner}](request)(handleResponseError)"
+        }
+
+        val requestTypology: Tree = request match {
+          case _: UnaryTpe =>
+            q"val request = Request[F](Method.$method, uri / ${uri.replace("\"", "")}).withEntity(req.asJson)"
+          case _: Fs2StreamTpe =>
+            q"val request = Request[F](Method.$method, uri / ${uri.replace("\"", "")}).withEntity(req.map(_.asJson))"
+          case _: MonixObservableTpe =>
+            q"val request = Request[F](Method.$method, uri / ${uri.replace("\"", "")}).withEntity(req.toReactivePublisher.toStream.map(_.asJson))"
+          case _ =>
+            q"val request = Request[F](Method.$method, uri / ${uri.replace("\"", "")})"
+        }
+
+        val responseEncoder =
+          q"""implicit val responseDecoder: EntityDecoder[F, ${response.safeInner}] = jsonOf[F, ${response.safeInner}]"""
+
+        def toRequestTree: Tree = request match {
+          case _: EmptyTpe =>
+            q"""def $name(implicit client: _root_.org.http4s.client.Client[F]): ${response.getTpe} = {
+		                  $responseEncoder
+		                  $requestTypology
+		                  $executionClient
+		                 }"""
+          case _ =>
+            q"""def $name(req: ${request.getTpe})(implicit client: _root_.org.http4s.client.Client[F]): ${response.getTpe} = {
+		                  $responseEncoder
+		                  $requestTypology
+		                  $executionClient
+		                 }"""
+        }
+
+        val routeTypology: Tree = (request, response) match {
+          case (_: Fs2StreamTpe, _: UnaryTpe) =>
+            q"""val requests = msg.asStream[${operation.request.safeInner}]
+              Ok(handler.${operation.name}(requests).map(_.asJson))"""
+
+          case (_: UnaryTpe, _: Fs2StreamTpe) =>
+            q"""for {
+              request   <- msg.as[${operation.request.safeInner}]
+              responses <- Ok(handler.${operation.name}(request).asJsonEither)
+            } yield responses"""
+
+          case (_: Fs2StreamTpe, _: Fs2StreamTpe) =>
+            q"""val requests = msg.asStream[${operation.request.safeInner}]
+             Ok(handler.${operation.name}(requests).asJsonEither)"""
+
+          case (_: MonixObservableTpe, _: UnaryTpe) =>
+            q"""val requests = msg.asStream[${operation.request.safeInner}]
+              Ok(handler.${operation.name}(Observable.fromReactivePublisher(requests.toUnicastPublisher)).map(_.asJson))"""
```

Comment:
I'm not pretty sure why are converting back and forth the streaming types (Observable <-> fs2.Stream).

Could we use `higherkindness.mu.rpc.internal.client.fs2Calls`, `higherkindness.mu.rpc.internal.server.fs2Calls`, `higherkindness.mu.rpc.internal.client.monixCalls` and `higherkindness.mu.rpc.internal.server.monixCalls`?

Reply:
For context, I created this part while we still used the old conversions for the RPC calls (before https://github.com/higherkindness/mu/pull/477), except that RPC had Observable as the underlying type and FS2 conversions to it, while HTTP had the reverse. So we might be able to leverage the new RPC approach here.

Reply:
We'll tackle this in the next iteration if both of you agree